src/ServiceAccount/Search/ClientIDFilter.tsx

```diff
@@ -0,0 +1,116 @@
+import { FilterProps } from './FilterProps';
+import React, { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TooltipContent } from './TooltipContent'
+import {
+  Button,
+  ButtonVariant,
+  InputGroup,
+  TextInput,
+  ToolbarFilter,
+  Tooltip,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
+
+export type ClientIDFilterProps = FilterProps;
+
+export const ClientIDFilter: React.FunctionComponent<ClientIDFilterProps> = ({
+  getSelectionForFilter,
+  onDeleteChipGroup,
+  onDeleteChip,
+  filterSelected,
+  isMaxFilter,
+  updateFilter,
+}) => {
+  const { t } = useTranslation();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const tooltipContent = TooltipContent(isMaxFilter, 'clientid');
+
+  const [value, setValue] = useState<string | undefined>();
+  const [valid, setValid] = useState<boolean>(true);
+
+  const validate = (value?: string) => {
+    return value ? !/["$^<>|+%/;:,\s*=~#()]/.test(value.trim()) : true;
+  };
+
+  const onFilter = () => {
+    if (value && value.trim() != '') {
+      if (validate(value)) {
+        updateFilter('clientid', { value: value, isExact: false }, false);
+        setValue('');
+      } else {
+        setValid(false);
+      }
+    }
+  };
+
+  const onKeyPress = (event) => {
+    if (event.key === 'Enter' && !isMaxFilter) {
+      onFilter();
+    }
+  };
+
+  const onChange = (input?: string) => {
+    setValue(input);
+    !valid && setValid(true);
+  };
+
+  const renderClientInput = () => {
+    if (filterSelected?.toLowerCase() === 'clientid') {
+      const v = !valid || isMaxFilter;
+
+      const FilterTooltip: React.FunctionComponent = () => {
```

Comment:
It's better not to create a component inside a render function. I remember reading a good article about why this is bad, but I can't find it, unfortunately. In short, this doesn't do what you'd expect it to do, which is to create a React component that behaves like a React component.

Since the render function will be called by React every time it needs to render the React tree - which is pretty often, every time some state change somewhere - everything that's defined inside the render function will be evaluated and executed every single time. This means that your `FilterTooltip` component gets "reinvented" every time the `ClientIDFilter` one gets rendered. The result is that the previously rendered `FilterTooltip` when compared with the newly rendered `FilterTooltip` will look as a different component to React, making it render it every single time even if not needed.

The same is true for anything done inside the render function. Callbacks are another thing that can cause unneeded renders. This is why callbacks should be wrapped inside [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback) unless they are static, meaning that they don't depend on anything coming as a prop.